orchestration/orchestration.go

```diff
@@ -180,15 +180,41 @@ func (o *Orchestrator) DeleteService(ctx context.Context, input *ServiceDeleteIn
 func (o *Orchestrator) processCluster(ctx context.Context, input *ServiceOrchestrationInput) (*ecs.Cluster, error) {
 	client := o.ECS
 	if input.Service.Cluster != nil {
-		log.Infof("Using provided cluster name %s", aws.StringValue(input.Service.Cluster))
+		log.Infof("Using provided cluster name %s", aws.StringValue(input.Cluster.ClusterName))
+
 		cluster, err := getCluster(ctx, client, input.Service.Cluster)
 		if err != nil {
 			return nil, err
 		}
+
 		log.Debugf("Got cluster %+v", cluster)
 		return cluster, nil
 	} else if input.Cluster != nil {
 		log.Infof("Creating cluster %s", aws.StringValue(input.Cluster.ClusterName))
+
+		tagCount := 0
+		for _, tag := range input.Cluster.Tags {
+			// the tag is set and not set to our Org, rewrite it
+			if aws.StringValue(tag.Key) == "spinup:tag" {
+				if aws.StringValue(tag.Value) != *aws.String(Org) {
+					tagCount++
+				}
+			}
+			// the tag is not set to our Org, rewrite it
+			if aws.StringValue(tag.Key) != "spinup:tag" {
+				tagCount++
+			}
+		}
+
+		//rewrite tag to our org
+		if tagCount > 0 {
+			input.Cluster.Tags = append(input.Cluster.Tags, &ecs.Tag{
+				Key:   aws.String("spinup:tag"),
+				Value: aws.String(Org),
+			})
+		}
```

Comment:
I dont think this code actually works?  It seems like if you have any tags you're just appending the org tag.  You should drop the iterator and create a tags slice initialized with the org tag before the loop, then append the loop if the key is not `spinup:org` or `yale:org`.  if the key is one of those org keys, skip it.

Reply:
updated - thx